pkg/cmd/pipeline/logs.go

```diff
@@ -0,0 +1,91 @@
+// Copyright © 2019 The Tekton Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipeline
+
+import (
+	"fmt"
+
+	"github.com/spf13/cobra"
+	"github.com/tektoncd/cli/pkg/cli"
+	"github.com/tektoncd/cli/pkg/cmd/pipelinerun"
+	"github.com/tektoncd/cli/pkg/helper/pods"
+	"github.com/tektoncd/pipeline/pkg/apis/pipeline/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func logCommand(p cli.Params) *cobra.Command {
+	opts := &pipelinerun.LogOptions{Params: p}
+	eg := `
+  # show the logs of the latest pipelinerun of given pipeline
+	tkn pipeline logs foo -n bar
+
+   `
+	c := &cobra.Command{
+		Use:                   "logs",
+		DisableFlagsInUseLine: true,
+		Short:                 "Show the logs of latest pipelinerun of given pipeline ",
+		Example:               eg,
+		Args:                  NameArg(p),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			opts.PipelineName = args[0]
+			opts.Stream = &cli.Stream{
+				Out: cmd.OutOrStdout(),
+				Err: cmd.OutOrStderr(),
+			}
+			opts.Streamer = pods.NewStream
+
+			if opts.PipelineName == "" {
+				return fmt.Errorf("missing mandatory argument pipeline")
+			}
+
+			if opts.PipelinerunName == "" {
+				cs, err := p.Clients()
+				if err != nil {
+					return err
+				}
+
+				opts.PipelinerunName, err = latestRunName(cs, p.Namespace())
+				if err != nil {
+					return err
+				}
+			}
+			return opts.Run()
+		},
+	}
+	c.Flags().BoolVarP(&opts.AllSteps, "all", "a", false, "show all logs including init steps injected by tekton")
+	c.Flags().BoolVarP(&opts.Follow, "follow", "f", false, "stream live logs")
+	c.Flags().StringVar(&opts.PipelinerunName, "pipeline-run", "pr", "show logs for mentioned pipelinerun only")
+	return c
+}
+
+func latestRunName(cs *cli.Clients, ns string) (string, error) {
+	runs, err := cs.Tekton.TektonV1alpha1().PipelineRuns(ns).List(metav1.ListOptions{})
```

Comment:
Is there any better method to get pipelineruns for a pipeline?

ATM, this code getting every pipelinerun from a namespace. Any upstream api?

@vdemeester @sthaha

Reply:
I just find out that we can use the label selector. Use that.

Reply:
@khrm yes using labels you can filter. `latestRunName` needs to use `opts.PipelineName` as paramater.

```go
func latestRunName(cs *cli.Clients, pipelineName, ns string) {
// […]
}
```

Reply:
Yes make sense if required at more than 2 places. :+1:

Reply:
Yes,  it makes sense.